### PR TITLE
[batch] prevent running startup script more than once

### PR DESCRIPTION
**Pull request by jigold**

### CODE REVIEW

**Review by daniel-goldstein — DISMISSED** [review]
How come we need to explicitly delete the instance and can't let the startup script run to completion? Having these potentially infinite loops in here makes me uneasy.

Regardless, the loop here is repeated at the end of the script, so I think we should at least move that into a `cleanup` function, add a `trap cleanup EXIT`, and then just `exit` in this conditional.

**Comment by jigold** [comment]
The reason for the while loop is the instance keeps running even if the startup script fails.

**Comment by jigold** [comment]
Also my bash is not real great, so definitely double check I got the function right.

**Comment by daniel-goldstein** [comment]
Ah I didn't notice that the two loops were in different scripts (startup vs. run). If the run-script exits cleanly will the worker shut down?

File: batch/batch/driver/create_instance.py [hunk 1]
```diff
@@ -109,7 +125,7 @@ async def create_instance(
                     'key': 'run_script',
                     'value': rf'''
 #!/bin/bash
-set -x
+set -ex
```
Comment (jigold) [inline batch/batch/driver/create_instance.py:128]: I'm not sure if we need this change.

Comment (daniel-goldstein) [reply batch/batch/driver/create_instance.py:128]: I think this is good we should fail early.

**Review by daniel-goldstein — APPROVED** [review]

**Pull request merged**